pkg/build/legacy/parser/deffile_test.go

```diff
@@ -0,0 +1,380 @@
+// Copyright (c) 2018-2019, Sylabs Inc. All rights reserved.
+// This software is licensed under a 3-clause BSD license. Please consult the
+// LICENSE.md file distributed with the sources of this project regarding your
+// rights to use or distribute this software.
+
+package parser
+
+import (
+	"bufio"
+	"encoding/json"
+	"errors"
+	"io/ioutil"
+	"os"
+	"reflect"
+	"strings"
+	"testing"
+
+	"github.com/sylabs/singularity/internal/pkg/test"
+	types "github.com/sylabs/singularity/pkg/build/legacy"
+)
+
+func TestScanDefinitionFile(t *testing.T) {
+	tests := []struct {
+		name     string
+		defPath  string
+		sections string
+	}{
+		{"Arch", "testdata_good/arch/arch", "testdata_good/arch/arch_sections.json"},
+		{"BusyBox", "testdata_good/busybox/busybox", "testdata_good/busybox/busybox_sections.json"},
+		{"Debootstrap", "testdata_good/debootstrap/debootstrap", "testdata_good/debootstrap/debootstrap_sections.json"},
+		{"Docker", "testdata_good/docker/docker", "testdata_good/docker/docker_sections.json"},
+		{"LocalImage", "testdata_good/localimage/localimage", "testdata_good/localimage/localimage_sections.json"},
+		{"Scratch", "testdata_good/scratch/scratch", "testdata_good/scratch/scratch_sections.json"},
+		{"Shub", "testdata_good/shub/shub", "testdata_good/shub/shub_sections.json"},
+		{"Yum", "testdata_good/yum/yum", "testdata_good/yum/yum_sections.json"},
+		{"Zypper", "testdata_good/zypper/zypper", "testdata_good/zypper/zypper_sections.json"},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, test.WithoutPrivilege(func(t *testing.T) {
+			deffile := tt.defPath
+			r, err := os.Open(deffile)
+			if err != nil {
+				t.Fatal("failed to read deffile:", err)
+			}
+			defer r.Close()
+
+			s := bufio.NewScanner(r)
+			s.Split(scanDefinitionFile)
+			for s.Scan() && s.Text() == "" && s.Err() == nil {
+			}
+
+			b, err := ioutil.ReadFile(tt.sections)
+			if err != nil {
+				t.Fatal("failed to read JSON:", err)
+			}
+
+			type DefFileSections struct {
+				Header string
+			}
+			var d []DefFileSections
+			if err := json.Unmarshal(b, &d); err != nil {
+				t.Fatal("failed to unmarshal JSON:", err)
+			}
+
+			// Right now this only does the header, but the json files are
+			// written with all of the sections in mind so that could be added.
+			if s.Text() != d[0].Header {
+				t.Fatal("scanDefinitionFile does not produce same header as reference")
+			}
+
+		}))
+	}
+}
+
+// Specific test to cover some corner cases of splitToken()
+//func TestSplitToken(t *testing.T) {
+//	ident_str := "test test1"
+//	content_str := "content1 content2"
+//	str := "%%%%" + ident_str + "\n" + content_str
+//	ident, content := splitToken(str)
+//	if ident != ident_str || content != content_str {
+//		t.Fatal("splitToken returned bad values")
+//	}
+//
+//	str = "%%" + ident_str
+//	ident, content = splitToken(str)
+//	if ident != ident_str || content != "" {
+//		t.Fatal("splitToken returned bad values")
+//	}
+//}
```

Comment:
Extra code?

Reply:
I'm not sure. @gvallee was this intended to be commented out when it was originally committed in #3084?